loopy/target/c/__init__.py

```diff
@@ -825,16 +828,23 @@ def get_function_definition(
                         tv.initializer is not None):
                     assert tv.read_only
 
-                    decl: Generable = self.wrap_global_constant(
+                    decl = self.wrap_global_constant(
                             self.get_temporary_var_declarator(codegen_state, tv))
 
                     if tv.initializer is not None:
-                        decl = Initializer(decl, generate_array_literal(
+                        init_decl = Initializer(decl, generate_array_literal(
                             codegen_state, tv, tv.initializer))
+                    else:
+                        init_decl = decl
 
-                    result.append(decl)
+                    result.append(init_decl)
+
+        assert isinstance(function_decl, FunctionDeclarationWrapper)
+        if not isinstance(function_body, Block):
+            function_body = Block([function_body])
 
         fbody = FunctionBody(function_decl, function_body)
```

Comment:
This `if` could be in cgen, but the current docs say that `FunctionBody` takes a block:
https://github.com/inducer/cgen/blob/738bdb6ea330de63dc1e476025b8ca693b535443/cgen/__init__.py#L992-L1006